### PR TITLE
[Process] Fix process continuing after reached timeout using getIterator()

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 3.1
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #20594 
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -307,7 +307,6 @@ public function start(callable $callback = null)
         }
 
         $this->updateStatus(false);
-        $this->checkTimeout();
```
Comment (chalasr) [inline src/Symfony/Component/Process/Process.php]: This change is not actually needed for fixing the bug, but I thought this line is useless.

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -558,6 +558,7 @@ public function getIterator($flags = 0)
             }
 
             $this->readPipesForOutput(__FUNCTION__, $blocking);
+            $this->checkTimeout();
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: should be above line 560, isn't it?

Comment (chalasr) [inline src/Symfony/Component/Process/Process.php]: true! fixed

**Comment by fabpot** [comment]
Thank you @chalasr.

**Pull request merged**